src/lib.rs

```diff
@@ -411,11 +448,35 @@ impl Datadog {
         tags: impl IntoIterator<Item = String>,
     ) {
         if self.is_reporting_enabled {
-            self.client.event(
-                metric.as_ref(),
-                text.as_ref(),
-                tags.into_iter().collect::<Vec<String>>(),
-            );
+            let tags = tags.into_iter().collect::<Vec<String>>();
+            let metric = metric.as_ref();
+            self.track_tags(metric, &tags);
+            self.client.event(metric, text.as_ref(), tags);
+        }
+    }
+
+    fn track_tags<'a>(&self, metric: impl AsRef<str>, tags: impl IntoIterator<Item = &'a String>) {
+        let hashes = tags
+            .into_iter()
+            .map(|t| {
+                let mut hasher = DefaultHasher::new();
+                hasher.write(metric.as_ref().as_bytes());
+                hasher.write(t.as_bytes());
+                hasher.finish()
+            })
+            .collect::<Vec<u64>>();
```

Comment:
I think this collect can be removed

Reply:
Are iterators lazy? I want to do as much of the hashing work outside the locked section as I can (it's probably not important but since I don't have a benchmark setup that simulates thread contention yet I figure better safe than sorry)

Actually you know what I'm going to figure out how to use criterion to simulate thread contention, that seems sensible, and then I can check if removing the collect has an impact

Reply:
Iterators in rust are lazy. I have not thought of this, I would leave a comment explaining that the collect is there to reduce the time spent when holding the lock. 
In that case you might also want to note that extending the hashmap unnecessarily hashes the final u64 hash. This could be prevented by implementing a custom noop hasher for u64. I'm not sure if reduced time spent holding the lock is significant enough to justify the effort though

Reply:
Yeah I thought about the double hash and was just kinda hoping hashset was doing something clever, but honestly it seems fine (in this case, hashing a single u64 should be an /awful/ lot cheaper than hashing an arbitrary str, I hope). It's somewhat much of a muchness, my thinking was just that hashing the tags is arbitrary amounts of work (since their length can vary), hashing the u64s is known. I'll add a comment